### ISSUE TITLE
Redraw current zoom level without viewer flashing

### ISSUE THREAD

**Issue by jwcharp** [openseadragon/openseadragon#1991]
Is it possible to redraw you current zoom level without the canvas getting cleared and flashing?

On the server I am updating a specific region of the tile source image by writing some text on it.
Since the only thing in the entire tile source that has change is just the section that has text on it.

I want to reload the current zoom level to display the new text.

I've used the following methods to clear the tile cache and redraw but the canvas flashes.
viewer.world.resetItems();
viewer.imageLoader.clear();
viewer.world.update();

If I just clear the the tile cache and reset the world, the text will show up if I change the zoom level.

Any advice would be greatly appreciated!

I am using a custom tile source similar to an image pyramid like DZI or Zoomify

**Comment by iangilman** [comment]
Interesting... I don't think we have a good mechanism for this. I think what you would want would be a way to tell the system to reload all of the tiles, but not to flush out the current tiles until the replacements have arrived. This would probably require updating OSD somehow. If you're interested in pursuing it (and ultimately making a pull request), I can try to point you in the right direction.

**Comment by jwcharp** [comment]
Thank you for your response,

I was able to find a workaround by adding a new tile in the viewer with an opacity of 0.01 then removing viewer(0) once viewer(1) was complete and setting viewer(0) - previously viewer(1) opacity to 1

viewer.addTiledImage({tileSource: tileSourceObject, opacity: 0.01});

The key was setting it to 0.01 if opacity was 0.00 looks like the tiles would not load

This seemed to work, however I am sure not optimal.

I am not so familiar with creating pull requests in GitHub, however I will try.

Any advice you can give to perhaps do this more elegantly would be helpful.

Thanks!

**Comment by iangilman** [comment]
@jwcharp yes, that's a great solution! If that works for you, then you're set. 

As for making a pull request to make this sort of thing smoother, I was thinking of something deeper than the solution you found. We would want to be able to tell a TiledImage to invalidate its cached tiles (so it loads them fresh) without removing the cached tiles (which is what it currently does) until you get the new ones to replace them. This would probably mean somehow modifying the chain of events that start with `TiledImage.reset`: 

https://github.com/openseadragon/openseadragon/blob/d04ccddebeb63a5645109c8f3cd4637848ed8bc4/src/tiledimage.js#L284

... and also having a more nuanced idea of `loaded` for Tiles (where one of the options would be 'invalidated' or something, meaning that it is available for drawing but should also be reloaded). Here's a good starting point for investigating that:

https://github.com/openseadragon/openseadragon/blob/d04ccddebeb63a5645109c8f3cd4637848ed8bc4/src/tiledimage.js#L1425

This is fairly in-depth stuff in the library, so don't feel like you need to take this on if you don't want to (especially since you already have a solution that works for you). I'm documenting it here so that anyone who wants to work on it has the info they need to get started. If that's you, that's great as well :)
